src/GraphQL/Execution/ExecutionContext.cs

```diff
@@ -39,5 +39,7 @@ public class ExecutionContext
         /// This can be useful for hiding error messages that reveal server implementation details.
         /// </summary>
         public Action<UnhandledExceptionContext> UnhandledExceptionDelegate { get; set; }
+
+        public IServiceProvider Services { get; set; }
```

Comment:
You motivate client code to use ServiceLocator antipatttern. See #1160 and discussion around it.

Reply:
From my point of view, not really.  I mean, the point of graphql is to access a back-end database.  Not static information like the StarWars sample.  So while the resolver function is immutable (which the schema and function definition still is), the database is mutable.  The usage implies that scope is _necessary_.  This is why ASP.Net creates a scope for each request - it just makes sense, regardless of the theory behind it.

Reply:
GraphQL is not about accessing databases. 

You expose `Services`. What will the client do with it? Call `GetService` method of course :)

>This is why ASP.Net creates a scope for each request - it just makes sense, regardless of the theory behind it.

Creating the scope and somehow accessing this scope are different things. The problem of resolving dependencies can be successfully solved without resorting to a ServiceLocator. The main thing is to allow the DI container to do the work for which it was created. I advise you to read the discussion in #1160.